alacritty_terminal/Cargo.toml

```diff
@@ -61,6 +61,8 @@ default = []
 live-shader-reload = []
 nightly = []
 bench = []
+# Use FreeType+FontConfig on all platforms, and enable HarfBuzz text shaping
+hb-ft = ["font/hb-ft"]
```

Comment:
The runtime option in it's current manifestation is harfbuzz and linux specific. This code is pretty narrow in scope in that it only works for harfbuzz and atm harfbuzz is only tested on linux. The scope can expand to cover other OSs but I would be wary to move this outside a feature flag while support on other OSes isn't in place.

Reply:
You can make it specific to `linux/bsd` rather than hided by "feature", which is perfectly fine as long as the requirements specified by @chrisduerr are met. Our font rendering will be feature less on macOs/Windows, but we shouldn't care about it for a now (e.g. we already has colored emojis on macOs, but not on linux). In fact we can use other libraries to make ligatures work on macOs/Windows.

> The runtime option in it's current manifestation is harfbuzz and linux specific.

FYI, we already has platform specific options in config (e.g.  `use_thin_strokes` for macOs).

Reply:
Having a feature flag on master is pretty much the same as not having the feature at all. In that case we could just keep the PR open until the reasons for having it as a feature flag are resolved.

However, for implementing something platform specific we can do precisely what @kchibisov said and use linux/bsd specific cfg tags.

However, even in that case we do want to make sure to reduce code duplication instead of just slapping feature/cfg attributes everywhere. The goal should be to use the same code for as much stuff is possible.